node/forest_libp2p/src/blocksync/provider.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use chain::Error as ChainError;
+use forest_blocks::{Tipset, TipsetKeys};
+use forest_cid::Cid;
+use ipld_blockstore::BlockStore;
+use log::debug;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use super::{
+    BlockSyncRequest, BlockSyncResponse, BlockSyncResponseStatus, CompactedMessages, TipsetBundle,
+};
+
+/// Builds blocksync response out of chain data.
+pub fn make_blocksync_response<DB>(db: Arc<DB>, request: &BlockSyncRequest) -> BlockSyncResponse
+where
+    DB: BlockStore,
+{
+    let mut response_chain: Vec<TipsetBundle> = vec![];
+
+    let mut curr_tipset_cids = request.start.clone();
+
+    loop {
+        let mut tipset_bundle: TipsetBundle = Default::default();
```

Comment:
```suggestion
        let mut tipset_bundle = TipsetBundle::default();
```
Little more readable imo

Reply:
updated